### PR TITLE
redundant_else pedantic check

### PR DESCRIPTION
**Pull request by tyshko5**
## Summary of changes

Changes introduced in this pull request:

The else block adds unnecessary indentation and verbosity especially in our cases when else block happens in the end of the function.

## Reference issue to close (if applicable)

Closes 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG](https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md) is up-to-date. All user-facing changes should be reflected in this document.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,8 +70,8 @@ lint: license clean
 	cargo fmt --all --check
 	taplo fmt --check
 	taplo lint
-	cargo clippy --features slow_tests --all-targets -- -D warnings -W clippy::unused_async
```
Comment (hanabi1224) [inline Makefile:73]: How about using `.clippy.toml`: https://doc.rust-lang.org/stable/clippy/configuration.html

**Comment by lemmih** [comment]
`redundant_else` is probably worthwhile but you need to explain why it should be enabled.

**Comment by tyshko5** [comment]
> redundant_else

The else block adds unnecessary indentation and verbosity especially in our cases when else block happens in the end of the function.

**Comment by lemmih** [comment]
> > redundant_else
> 
> The else block adds unnecessary indentation and verbosity especially in our cases when else block happens in the end of the function.

Your arguments belong in the PR description, not in the comments.

**Review by LesnyRumcajs — APPROVED** [review]

**Review by lemmih — APPROVED** [review]

**Pull request merged**